specification/sbi_atee.adoc

```diff
@@ -13,6 +44,953 @@ running a VMM).
+. Mapping demand-zero confidential memory regions
+. Adding confidential VCPUs to the TVM
+. Finalizing TVM creation
+. Scheduling TVM execution
+. Management of secure interrupts
+. Handling and servicing TVM faults and exits
+. Mapping of confidential zero pages on demand
+. Mapping non-confidential shared pages on demand
+. Tearing down TVMs
+. Repurposing confidential memory for other TVMs
+. Reclaiming confidential memory for non-confidential VMs
+
+=== Platform TSM detection and capability enumeration
+
+Platform support for the TSM can be detected by probing for the EXT_TEE extension, and then
+calling sbi_tee_tsm_get_info() to get information about the current status of the TSM. The
+TSM must be be in TSM_READY in order to process further ECALLs.
+
+=== Conversion of non-confidential memory to confidential memory
+
+Platform memory is non-confidential by default, and must be converted to confidential memory
+before use with TVMs. The conversion process is initiated by designating the host physical
+pages that to be converted, and then issuing fence operations to ensure that all outstanding
+TLB entries to the non-confidential memory are flushed across all VCPUs on the platform. This
+ensures that there's no overlapping mapping between the confidential and non-confidential memory
+regions on the platform.
+
+This requires the host to make three separate ECALLs to the TSM:
+
+. sbi_tee_tsm_convert_pages()
+. sbi_tee_tsm_initiate_fence()
+. sbi_tee_tsm_local_fence()
+
+The memory conversion process is complete when sbi_tee_tsm_local_fence() is successfully completed
+on the every VCPU on the platform.
+
+Converted memory may be assigned to TVMs, but cannot be repurposed for regular use unless it's reclaimed.
+
+=== TVM creation
+
+TVMs are created using the sbi_tee_create_tvm(). This creates a TVM, and sets with state "initializing". The
+host must assign confidential memory for page tables, payload mapping, and VCPUs before it can be transitioned
+into a "runnable" state.
+
+=== TVM memory management
+
+The host is responsible for the following:
+
+. Dontations for the page-table pool
+.. The host must ensure that the TSM has sufficient confidential memory for mapping and
+   managing TVM page-tables for the code and data payloads (sbi_tee_tvm_add_page_table_pages())
+. Defining memory regions for confidential, MMIO, and shared memory regions
+.. The host must declare the TVM physical address ranges for mapping of confidential and
+   non-confidential memory. There can be multiples ranges of each type, but no two regions
+   can overlap. Both confidential and non-confidential regions can be sparsely mapped; however,
+   any sparsely mapped confidential page that's demand-page can only be a demand-zero page. 
+   The `sbi_tee_tvm_add_confidential_memory_region()`, `sbi_tee_tvm_add_emulated_mmio_region()`,
+   and `sbi_tee_tvm_add_shared_memory_region()` can be used for this purpose.
```

Comment:
I suggested earlier to combine these three into one function and pass a type parameter. Any thoughts on that ?